### PR TITLE
Rename and move _JSONSpeechAPI to HTTPSpeechAPI.

### PR DESCRIPTION
**Pull request by daspecster**
Moves `_JSONSpeechAPI` out of `client.py` and into `_http.py`. 

Also renames `_JSONSpeechAPI` -> `HTTPSpeechAPI`.

#2965 needs to be merged before this.

### CODE REVIEW

File: speech/google/cloud/speech/_http.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
+# Copyright 2017 Google Inc.
```
Comment (dhermes) [inline speech/google/cloud/speech/_http.py]: 2017! ZOMG

File: speech/google/cloud/speech/_http.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                                   profanity_filter, speech_context)
+        api_response = self._connection.api_request(
+            method='POST', path='speech:syncrecognize', data=data)
+
+        if len(api_response['results']) == 1:
+            result = api_response['results'][0]
+            return [Alternative.from_api_repr(alternative)
+                    for alternative in result['alternatives']]
+        else:
+            raise ValueError('More than one result or none returned from API.')
```
Comment (dhermes) [inline speech/google/cloud/speech/_http.py]: Does this match the gRPC impl? (I realize you just cut-and-pasted this.)

File: speech/google/cloud/speech/_http.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                           used to improve the accuracy for specific words
+                           and phrases. This can also be used to add new
+                           words to the vocabulary of the recognizer.
+
+    :rtype: dict
+    :returns: Dictionary with required data for Google Speech API.
+    """
+    if sample.content is not None:
+        audio = {'content':
+                 _bytes_to_unicode(b64encode(_to_bytes(sample.content)))}
```
Comment (dhermes) [inline speech/google/cloud/speech/_http.py]: This triply-nested call is icky (though I realize why it's necessary)

File: speech/google/cloud/speech/_http.py (hunk 2 above)
Comment (daspecster) [inline speech/google/cloud/speech/_http.py]: Actually #2965 needs to be merged in before this PR.

File: speech/google/cloud/speech/_http.py (hunk 3 above)
Comment (daspecster) [inline speech/google/cloud/speech/_http.py]: yeah ick.

File: speech/google/cloud/speech/_http.py (hunk 1 above)
Comment (daspecster) [inline speech/google/cloud/speech/_http.py]: 🎆

**Pull request merged**